pkg/reconciler/namespace/namespace.go

```diff
@@ -211,11 +225,15 @@ func (c *Controller) isValidCluster(ctx context.Context, lclusterName, clusterNa
 	if err != nil {
 		return false, err
 	}
-	isReady := conditions.IsTrue(cluster, clusterv1alpha1.ClusterReadyCondition)
-	if !isReady {
+	if ready := !conditions.IsTrue(cluster, clusterv1alpha1.ClusterReadyCondition); !ready {
 		klog.Infof("Cluster %q|%q is not reporting ready", lclusterName, clusterName)
+		return false, nil
 	}
-	return isReady, nil
+	if evictAfter := cluster.Spec.EvictAfter; evictAfter != nil && evictAfter.Time.Before(time.Now()) {
+		klog.Infof("Cluster %q|%q is evicted", lclusterName, clusterName)
```

Comment:
nit: s/evicted/cordoned/ ? i.e. namespaces scheduled to a namespace are evicted. clusters are not evicted. 

Maybe this suggests using `cordonAfter`? There is ambiguity to me in what `cluster.evictAfter` means, since the natural implication for me is that the cluster should be evicted. `cluster.cordonAfter` has no such ambiguity.

Reply:
Good point, about clusters not being evicted, only namespaces/workloads.

However, `evictAfter` != `cordonAfter` -- cordoning affects only new workloads making a scheduling decision, evicting affects already-scheduled workloads, kicking them out. Maybe `drainAfter`?

(Now I'm thinking maybe it's worth expressing `unschedulableAfter`/`cordonAfter` to be consistent with `evictAfter` -- or maybe we should just do whatever we can now, and aim to replace all of this with an actual scheduling constraint language at some point in the future?)

Reply:
The hardest task of software engineering: naming ;)

Maybe `unscheduleAfter` to be specific about what is being controlled? But I digress. I don't want to block merge of this on naming.